src/Aspire.Hosting.Apache.Pulsar/PulsarBuilderExtensions.cs

```diff
@@ -0,0 +1,109 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire.Hosting.Apache.Pulsar;
+using Aspire.Hosting.ApplicationModel;
+using Aspire.Hosting.Utils;
+
+namespace Aspire.Hosting;
+
+/// <summary>
+/// Provides extension methods for adding Pulsar resource to a <see cref="IDistributedApplicationBuilder"/>.
+/// </summary>
+public static class PulsarBuilderExtensions
+{
+    /// <summary>
+    /// Adds Pulsar container to the application module
+    /// Runs in standalone mode by default.
+    /// </summary>
+    /// <remarks>
+    /// The default image and tag are "apachepulsar/pulsar" and "3.2.0".
+    /// </remarks>
+    /// <param name="builder">The <see cref="IDistributedApplicationBuilder"/>.</param>
+    /// <param name="name">The name of the resource. This name will be used as the endpoint name when referenced in dependency.</param>
+    /// <param name="servicePort">The service port that the underlying container is bound to when running locally.</param>
+    /// <param name="brokerPort">The broker port that the underlying container is bound to when running locally.</param>
+    /// <returns>A reference to the <see cref="IResourceBuilder{T}"/>.</returns>
+    public static IResourceBuilder<PulsarResource> AddPulsar(
+        this IDistributedApplicationBuilder builder,
+        string name,
+        int? servicePort = null,
```

Comment:
we use `targetPort` more consistently to refer to the underlying container port and `port` for the broker port

Reply:
Renamed `servicePort` to `targetPort` and `brokerPort` to `port`. Updated comments accordingly.